solid/src/main/java/com/inrupt/client/solid/SolidClientException.java

```diff
@@ -107,6 +116,62 @@ public String getBody() {
         return body;
     }
 
+    /**
+     * It is acceptable for a SolidClient instance to be in a classpath without any implementation for
+     * JsonService, in which case the ProblemDetails exceptions will fallback to default and not be parsed.
+     * @return a JsonService instance, null if none is available
+     */
+    private static JsonService getJsonService() {
+        if (SolidClientException.jsonServiceInitialized) {
+            return SolidClientException.jsonService;
+        }
+        JsonService js;
+        try {
+
+            js = ServiceProvider.getJsonService();
+        } catch (IllegalStateException e) {
+            js = null;
+        }
+        SolidClientException.jsonService = js;
+        SolidClientException.jsonServiceInitialized = true;
+        return SolidClientException.jsonService;
+    }
+
+    public static SolidClientException fromErrorResponse(
```

Comment:
Arguably, moving this method here makes it possible to make the `handle` method private.